src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -648,19 +1114,97 @@ public static SyntaxToken XmlTextLiteral(SyntaxTriviaList leading, string text,
+            return new SyntaxToken(
+                InternalSyntax.SyntaxFactory.XmlTextNewLine(
+                    (InternalSyntax.CSharpSyntaxNode)leading.Node,
+                    text,
+                    value,
+                    (InternalSyntax.CSharpSyntaxNode)trailing.Node));
         }
 
         /// <summary>
-        /// Creates a token with kind XmlEntityLiteralToken.
+        /// Creates the syntax representation of an xml newline token for xml documentation comments.
+        /// </summary>
+        /// <param name="text">The raw text within the new line.</param>
+        /// <param name="continueXmlDocumentationComment">
+        /// If set to true, a documentation comment exterior token will be added to the trailing trivia
+        /// of the new token.</param>
+        public static SyntaxToken XmlTextNewLine(string text, bool continueXmlDocumentationComment)
+        {
+            var token = new SyntaxToken(
+                InternalSyntax.SyntaxFactory.XmlTextNewLine(
+                    (InternalSyntax.CSharpSyntaxNode)ElasticMarker.UnderlyingNode, 
+                    text, 
+                    text, 
+                    (InternalSyntax.CSharpSyntaxNode)ElasticMarker.UnderlyingNode));
+
+            if (continueXmlDocumentationComment)
+                token = token.WithTrailingTrivia(token.TrailingTrivia.Add(DocumentationCommentExterior("/// ")));
+
+            return token;
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of a xml text node (e.g. for xml documentation comments).
+        /// </summary>
+        /// <param name="value">The string literal used as the text of the xml text node.</param>
+        public static XmlTextSyntax XmlText(string value)
+        {
+            return XmlText(XmlTextLiteral(value));
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of a xml text node (e.g. for xml documentation comments).
+        /// </summary>
+        /// <param name="textTokens">A list of text tokens used as the text of the xml text node.</param>
+        public static XmlTextSyntax XmlText(params SyntaxToken[] textTokens)
+        {
+            return XmlText(TokenList(textTokens));
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of an xml text literal.
+        /// </summary>
+        /// <param name="value">The text used within the xml text literal.</param>
+        public static SyntaxToken XmlTextLiteral(string value)
+        {
+            // TODO: [RobinSedlaczek] It is no compiler hot path here I think. But the contribution guide
+            //       states to avoid LINQ (https://github.com/dotnet/roslyn/wiki/Contributing-Code). With
+            //       XText we have a reference to System.Xml.Linq. Isn't this rule valid here? 
```

Comment:
Yes, the ban is about not using LINQ queries in hot paths like parser and binder, because of the allocations.  But they get used a lot in IDE layers.